Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm

```diff
@@ -294,6 +295,25 @@
         context->didFailLoadForFrame(pageID, frameID, parentFrameID, frameURL, timestamp);
 }
 
+void WebExtensionController::handleContentRuleListNotification(WebPageProxyIdentifier pageID, URL& url, WebCore::ContentRuleListResults& results)
+{
+    for (const auto& result : results.results) {
+        auto contentRuleListIdentifier = result.first;
+        for (auto& context : m_extensionContexts) {
+            if (context->uniqueIdentifier() != contentRuleListIdentifier)
+                continue;
+
+            RefPtr tab = context->getTab(pageID);
+            if (!tab)
+                break;
```

Comment:
You should move this out of the loop, since it is the same for all results.

Reply:
Is it? Getting the tab seems to require the context, and what happens if one specific extension doesn't have access to a tab?

If they are the same, then I'm fine moving this out of the loop - how would you recommend getting the tab for one particular extension?